rotkehlchen/chain/evm/decoding/gearbox/gearbox_cache.py

```diff
@@ -0,0 +1,305 @@
+        verified_tokens: defaultdict[str, set[str]] = defaultdict(set)
+        for token_type, token_addresses in zip(('pools', 'farm'), (pool.lp_tokens, [pool.farming_pool_token]), strict=False):  # noqa: E501
+            for token_address in token_addresses:
+                try:
+                    token = get_or_create_evm_token(
+                        userdb=evm_inquirer.database,
+                        evm_address=string_to_evm_address(token_address),
+                        chain_id=evm_inquirer.chain_id,
+                        protocol=CPT_GEARBOX,
+                        evm_inquirer=evm_inquirer,
+                        underlying_tokens=(
+                            [UnderlyingToken(address=pool.underlying_token, token_kind=EvmTokenKind.ERC20, weight=ONE)]  # noqa: E501
+                            if pool.underlying_token is not None else []
+                        ),
+                        encounter=TokenEncounterInfo(
+                            description='Querying gearbox pools',
+                            should_notify=False,
+                        ),
+                    )
+                    verified_tokens[token_type].add(token.identifier)
+                except NotERC20Conformant as e:
+                    log.error(
+                        f'Skipping pool {pool} because {token_address} is not a '
+                        f'valid ERC20 token. {e}',
+                    )
+                    continue
+
+        pool.farming_pool_token = verified_tokens['farm'].pop()
+        pool.lp_tokens = verified_tokens['pools']
+        verified_pools.append(pool)
+    return verified_pools
+
+
+def get_zapper_tokens(inquirer: 'EvmNodeInquirer', pool_data: list[str], underlying_token_address: ChecksumEvmAddress) -> set[ChecksumEvmAddress] | None:  # noqa: E501
+    """Extracts zapper tokens from the pool data, excluding the underlying token address
+    and the special ETH address."""
+    try:
+        return {
+            de_addr
+            for data in pool_data[23]  # ZapperInfo[]
+            for addr in data[1:]  # ignore zapper address, only consider tokenIn and tokenOut
+            if (de_addr := deserialize_evm_address(addr)) not in {underlying_token_address, ETH_SPECIAL_ADDRESS}  # noqa: E501
+        }
+    except (DeserializationError, IndexError) as e:
+        log.error(f'Could not deserialize zapper tokens from {inquirer.chain_name} gearbox pool data with underlying as {underlying_token_address}: {e}')  # noqa: E501
+        return None
+
+
+def query_gearbox_data_from_chain(
+        evm_inquirer: 'EvmNodeInquirer',
+        existing_pools: set[ChecksumEvmAddress],
+) -> list[GearboxPoolData] | None:
+    """
+    Query all Gearbox information(lp tokens, pools, lp coins) from data compressor.
+
+    May raise:
+    - RemoteError if failed to query chain
+    """
+    data_compressor = evm_inquirer.contracts.contract(GEARBOX_DATA_COMPRESSOR)
+    pools_data = data_compressor.call(
```

Comment:
don't see it used in any other place

```suggestion
    pools_data = evm_inquirer.contracts.contract(GEARBOX_DATA_COMPRESSOR).call(
```